publication-event-handlers/src/main/java/no/unit/nva/publication/events/handlers/tickets/AcceptedPublishingRequestEventHandler.java

```diff
@@ -180,121 +173,56 @@ private RuntimeException throwException(String message, Failure<?> failure) {
-                                                                                 publishingRequestCase)
-                                                                : pendingInternalFile;
-            case PendingOpenFile pendingOpenFile -> publishingRequestCase
-                                                        .getApprovedFiles()
-                                                        .contains(pendingOpenFile.getIdentifier())
-                                                        ? toOpenFile(pendingOpenFile, publishingRequestCase)
-                                                        : pendingOpenFile;
+            case UnpublishedFile unpublishedFile when publishingRequest.fileIsApproved(unpublishedFile) ->
+                toPublishedFile(unpublishedFile, publishingRequest);
+            case PendingInternalFile pendingInternalFile when publishingRequest.fileIsApproved(pendingInternalFile) ->
+                toInternalFile(pendingInternalFile, publishingRequest);
+            case PendingOpenFile pendingOpenFile when publishingRequest.fileIsApproved(pendingOpenFile) ->
+                toOpenFile(pendingOpenFile, publishingRequest);
             case null, default -> associatedArtifact;
         };
     }
 
-    private static void logFilePublish(File unpublishedFile, PublishingRequestCase publishingRequestCase) {
-        logger.info(
-                PUBLISHING_FILE_MESSAGE,
-                unpublishedFile.getIdentifier(),
-                unpublishedFile.getClass().getSimpleName(),
-                publishingRequestCase.getIdentifier(),
-                publishingRequestCase.getResourceIdentifier());
-    }
-
-    private static PublishedFile toPublishedFile(
-            UnpublishedFile unpublishedFile, PublishingRequestCase publishingRequestCase) {
-        logFilePublish(unpublishedFile, publishingRequestCase);
-        return unpublishedFile.toPublishedFile();
-    }
-
-    private static InternalFile toInternalFile(
-            PendingInternalFile pendingInternalFile, PublishingRequestCase publishingRequestCase) {
-        logFilePublish(pendingInternalFile, publishingRequestCase);
-        return pendingInternalFile.toInternalFile();
-    }
-
-    private static OpenFile toOpenFile(
-        PendingOpenFile pendingOpenFile, PublishingRequestCase publishingRequestCase) {
-        logFilePublish(pendingOpenFile, publishingRequestCase);
-        return pendingOpenFile.toOpenFile();
-    }
-
-    private Publication fetchPublication(
-            UserInstance userInstance, SortableIdentifier publicationIdentifier) {
-        return attempt(() -> resourceService.getPublication(userInstance, publicationIdentifier))
-                .orElseThrow();
+    private Publication fetchPublication(SortableIdentifier publicationIdentifier) {
+        return attempt(() -> resourceService.getPublicationByIdentifier(publicationIdentifier)).orElseThrow();
```

Comment:
This line fixes the issue. When fetching publication by using userInstance, it should match resourceOwner of publication. In our case it can differ, since ticket can be owned by other institution than publication.